src/steps/AddAnother.js

```diff
@@ -105,12 +105,19 @@ class AddAnother extends Question {
     } else if (req.method === 'POST') {
       this.parse();
       this.validate();
-
       if (this.valid) {
         this.store();
-        res.redirect(this.path);
+        if (this.xhr) {
+          res.status(OK).json({ validationErrors: [] });
+        } else {
+          res.redirect(this.path);
+        }
       } else {
-        res.render(this.template, this.locals);
+        if (this.xhr) {
+          res.json(this.buildValidationErrors);
```

Comment:
@bendiggle can you pls change this to 
`res.status(UNPROCESSABLE_ENTITY).json(er);` -- I need it :)

(btw you also need to import the thingie at the top)

Reply:
@barbaracassani If I do this it means all the logic for our field validation within the ajax would have to be in the errorHandler rather than the successHandler. I'm not sure that's what we want

Reply:
But that's the point exactly -- so you can have a neat callback for the error and one for the success. Which means that most request libraries will handle that for you.

Reply:
Also, it's more semantic.

Reply:
And if / when we use Promises, it's soooo straightforward to have `then` and `reject` depending on the status code.

Reply:
Pleeeeeaseeeeee :)

Reply:
You win